cmake/helpers.cmake

```diff
@@ -137,8 +137,10 @@ macro(add_sanitizer_flag flag)
 
     # Check C and CXX compilers for given sanitizer flag.
     check_c_compiler_flag("${SANITIZER_FLAG}" "C_${check_name}")
-    check_cxx_compiler_flag("${SANITIZER_FLAG}" "CXX_${check_name}")
-    if (${C_${check_name}} OR ${CXX_${check_name}})
+    if (CMAKE_CXX_STANDARD_REQUIRED)
+        check_cxx_compiler_flag("${SANITIZER_FLAG}" "CXX_${check_name}")
+    endif()
+    if (${C_${check_name}} OR $<BOOL:CXX_${check_name}>)
```

Comment:
don't you need this `BOOL` a few lines below as well...?

Reply:
It's not needed at all after making some changes.